scanpy/tests/test_plotting.py

```diff
@@ -1215,6 +1215,57 @@ def test_scatter_specify_layer_and_raw():
         sc.pl.umap(pbmc, color="HES4", use_raw=True, layer="layer")
 
 
+@pytest.fixture
+def pbmc_filtered():
+    pbmc = sc.datasets.pbmc68k_reduced()
+    sc.pp.filter_genes(pbmc, min_cells=10)
+    return pbmc
+
+
+@pytest.mark.parametrize(
+    "x,y,color,use_raw,value_error",
+    [
+        # test that plots work with var_names only found in raw if use_raw
+        # is None or True
+        ('EGFL7', 'F12', 'FAM185A', None, False),
+        ('EGFL7', 'F12', 'FAM185A', True, False),
```

Comment:
These should be equivalent, right?

Could cases like this instead be tested using `check_same_image` where:

```python
sc.pl.scatter(pbmc, x='EGFL7', y='F12', color='FAM185A', use_raw=None)
# is equivalent to:
sc.pl.scatter(pbmc, x='EGFL7', y='F12', color='FAM185A', use_raw=True)
# and also to:
sc.pl.scatter(pbmc.raw.to_adata() x='EGFL7', y='F12', color='FAM185A')
```

And then another case where you only use values from `obs`?

Reply:
Yes, good idea, see bc949924704ff149e2677c6b68adab6c0eadb353

Reply:
However, this does not work for `obs`, before or after this PR. This works as expected:
```python
sc.pl.scatter(pbmc, x="AAAGCCTGGCTAAC-1", y="AAATTCGATGCACA-1", use_raw=False)
```
but this doesn't:
```python
sc.pl.scatter(pbmc, x="AAAGCCTGGCTAAC-1", y="AAATTCGATGCACA-1", use_raw=True)
```
because `sc.pl.scatter()` transposes `pbmc` in both cases, but `raw` doesn't get transposed, resulting in `AttributeError: 'NoneType' object has no attribute 'obs_vector'` because `pbmc.raw` gets wiped out by the transposition.

This is a different bug than the one this PR fixes. I can make a separate issue and work on this separately if you agree that this isn't the correct behavior.

Reply:
Yeah, I think that should work. Would appreciate the PR!